python/dgl/frame.py

```diff
@@ -548,7 +559,17 @@ def record_stream(self, stream):
         """
         if F.get_preferred_backend() != "pytorch":
             raise DGLError("record_stream only supports the PyTorch backend.")
-        self.data.record_stream(stream)
+        # Only record streams for materialized tensors otherwise just record for the index
+        if self.index is None:
+            self.data.record_stream(stream)
```

Comment:
Do we still need recording stream on `data`?

Reply:
no we can still record on the `storage`. I just pushed a new commit. Let me know if that looks good.